lib/stdlib/src/edlin_expand.erl

```diff
@@ -20,43 +20,780 @@
 -module(edlin_expand).
 
 %% a default expand function for edlin, expanding modules and functions
-
--export([expand/1, format_matches/1]).
+-include("../../kernel/include/eep48.hrl").
+-export([expand/1, expand/3, format_matches/1, get_exports/1]).
 
 -import(lists, [reverse/1, prefix/2]).
 
-%% expand(CurrentBefore) ->
-%%	{yes, Expansion, Matches} | {no, Matches}
+%% Will remove this before delivery
+%% Questions:
+%% Types:
+%%  should we print all the leaf concrete types or stop at user_type and remote_type?
+%%   - file:open("/", )
+%% Maps:
+%%  cant support #{hej:=T} = M because we cannot look ahead to see that its M we extract from
+%%      could try to complete with all keys of all maps in Bindings list,
+%%      but it could just confuse the user, and chances are that most of the time the user
+%%      is interested in extracting a key of a map that is returned from a function
+%% Hidden functions:
+%%  should we remove them from display, but still support autocompletion of them?
+%% Imported functions / shell commands (available in shell_default module):
+%%  should we make these visible for auto completion, is_atom/1, ht ...
+%%  should a banner for different suggestions be shown when suggestions are of different types?
```

Comment:
Actually, maybe we can do it like this:

```
1> y<TAB>
y/1           y/2           yecc:
yeccparser:   yeccscan:
```

and then only autocomplete things in shell_default as top level expressions, so if we do:

```
1> element(y<TAB>
yecc:         yeccparser:   yeccscan:
```
it would only show the modules.

The difference between auto-imported functions (such as is_atom) and shell_default can be left for a potential syntax highlighter.

Reply:
It is done,

I have been thinking a little bit on formatting please have a look and let me know your preferences
https://gist.github.com/frazze-jobb/5ea36dd14a8115b06e670c7f1e79f18f

additionally,
should we special handle file:filename_all(), file:name_all() and just print filename()
currently date() has a range type printing 1..31
do we want to write non_neg_integer() as 0..
neg_integer() ..-1
positive_integer() 1..
or just keep them?

Reply:
I also wrote some comments in the gist.

> should we special handle file:filename_all(), file:name_all() and just print filename()

Do you mean instead of expanding the type to its base types? If so, then yes I would say that we want that. If we cannot come up with a general method that does what we want.

> currently date() has a range type printing 1..31

Maybe we can update the type to include a name? i.e. 

```erlang
-type date()           :: {Year :: year(),Month :: month(),Day :: day()}.
```

Or do we want to also include the name of the expanded type. i.e. 

```bash
1> calendar:date_to_gregorian_days({<TAB>
year() :: 1970..10000
```

> do we want to write non_neg_integer() as 0..
> neg_integer() ..-1
> positive_integer() 1..

In erl_docgen we translate these to be `integer() >= 0`: https://github.com/erlang/otp/blob/master/lib/erl_docgen/src/docgen_otp_specs.erl#L279-L290. So to be consistent maybe we should do the same here?